lib/spack/llnl/util/filesystem.py

```diff
@@ -282,6 +284,60 @@ def working_dir(dirname, **kwargs):
 
 
 @contextmanager
+def replace_directory_transaction(directory_name, tmp_root=None):
+    """Moves a directory to a temporary space. If the operations executed
```

Comment:
This PR looks all good except I have one minor request: I think the following reword would be clearer:

> If the operations executed within the context manager don't raise an exception, the directory is deleted. If there is an exception, the move is undone.

Reply:
done